check_limits.py

```diff
@@ -1,18 +1,43 @@
 
-def battery_is_ok(temperature, soc, charge_rate):
+def battery_is_ok(temperature, soc, charge_rate):     
+  status_list = [temperature_in_range(temperature), soc_in_range(soc), charge_rate_in_range(charge_rate)]
+  for statuses in status_list:
+    if statuses == False:
+      return False
+  return True
+
+
+def temperature_in_range(temperature):
   if temperature < 0 or temperature > 45:
     print('Temperature is out of range!')
     return False
-  elif soc < 20 or soc > 80:
+  return True
+
+
+def soc_in_range(soc):
+  if soc < 20 or soc > 80:
```

Comment:
usage of hard coded values will lead to issues during maintenance. if it gets used in many places and then a change in one place needs to be consistent with all others. There is a scope for improvement here.